pkg/ocicni/ocicni.go

```diff
@@ -144,9 +143,7 @@ func newWatcher(confDir string) (*fsnotify.Watcher, error) {
 	return watcher, nil
 }
 
-func (plugin *cniNetworkPlugin) monitorConfDir(start *sync.WaitGroup) {
-	start.Done()
-	plugin.done.Add(1)
```

Comment:
plugin.done.Add within a goroutine is not thread safe. preserved on line 248.

Reply:
If sync.WaitGroup isn't thread safe, then we have giant problems everywhere. AFAIK it is threadsafe to use from goroutines and in fact internally it uses `atomic` operations like AddUint64.